### PR TITLE
Remove actor#submit from ScheduleService

### LINKED ISSUE #10291: Remove the additional submit
**Description**

In the ProcessingScheduleServiceImpl we have a method, [which submits a task to our scheduling queue](https://github.com/camunda/zeebe/blame/main/engine/src/main/java/io/camunda/zeebe/streamprocessor/ProcessingScheduleServiceImpl.java#L71) (before it will be really submitted, like for run delayed).

This is somehow unexpected and can cause race conditions (it will slow down the execution). It looks like it was added only for testability https://github.com/camunda/zeebe/commit/f99a9a46a0c32eba51d0b5d3942a7014d6243209 But we can achieve this also differently, by using an Actor in our tests.

### PR DESCRIPTION
**Pull request by ChrisKujawa**
## Description

After https://github.com/camunda/zeebe/pull/10468 is merged we can open the PR for review.

### PR contains:

* Use an own executor in the EngineRule for the exporting of the records, instead of the ProcessingScheduleService. The service should only be accessed by the same actor (Processing Actor).
* Remove LogStream from context (finally)
* Remove [remove ProcessingScheduleServiceIntegrationTest](https://github.com/camunda/zeebe/commit/88fb590f2e8b73e3314f769173849b156bb3a383) this is no longer possible to test like this, as written above the service should only accessed by the same actor. We have as replacement the ProcessingScheduleServiceTest which uses an own actor for the test.
* Remove FINALLY the extra actor#submit

## Related issues

closes #10291 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by Zelldon** [comment]
@oleschoenburg failing test is fixed via https://github.com/camunda/zeebe/pull/10573

File: engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java [hunk 1]
```diff
@@ -423,28 +427,43 @@ private static class ProcessingExporterTransistor implements StreamProcessorLife
...
+      // this is called from outside (LogStream), so we need to enqueue the task
+      if (executorService.isShutdown()) {
+        return;
+      }
+
+      executorService.submit(this::exportEvents);
+    }
+
+    private void exportEvents() {
+      // to always run in the same thread
```
Comment (oleschoenburg) [inline engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java:466]: :wrench: looks like the comment is a left-over.

File: engine/src/main/java/io/camunda/zeebe/engine/api/ReadonlyStreamProcessorContext.java [hunk 2]
```diff
@@ -7,18 +7,10 @@
...
-
 public interface ReadonlyStreamProcessorContext {
 
   ProcessingScheduleService getScheduleService();
 
-  /**
-   * @return the logstream, on which the processor runs
-   */
-  @Deprecated // only used in EngineRule; TODO remove this
-  LogStream getLogStream();
```
Comment (oleschoenburg) [inline engine/src/main/java/io/camunda/zeebe/engine/api/ReadonlyStreamProcessorContext.java:20]: :heart_eyes:

**Review by lenaschoenburg — APPROVED** [review]
Nice, looks good :+1: 

:wrench: I think `ProcessingScheduleServiceImpl` could be package-private now if we move `ProcessingScheduleServiceTest` to the streamprocessor package too. Also, the docs there could be updated: "Here the implementation is just a suggestion" sounds outdated :smile:

**Comment by Zelldon** [comment]
Thanks for the fast review @oleschoenburg :rocket: 

If you are ok with it I would like to keep the visibility for now, since I have to move classes anyway until I'm done with https://github.com/camunda/zeebe/issues/10130. Example PR https://github.com/camunda/zeebe/pull/10454 (which is upcoming).

File: engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java (hunk 1 above)
Comment (oleschoenburg) [reply engine/src/test/java/io/camunda/zeebe/engine/util/EngineRule.java:466]: Ah okay, I thought you experimented with submitting to the executor here and that's where this comment got introduced. Then maybe change the comment to "skip if not yet recovered" or something along those lines?

**Comment by oleschoenburg** [comment]
> If you are ok with it I would like to keep the visibility for now

Sure, was just a suggestion anyway :+1:

**Comment by Zelldon** [comment]
Bors r+

**Pull request merged**